app/Parser/NodeList.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+namespace DevCommunityDE\CodeFormatter\Parser;
+
+use IteratorAggregate;
+
+final class NodeList implements IteratorAggregate
+{
+    /** @var Node[] */
+    private $nodes;
+
+    /**
+     * constructs a new node-list.
+     *
+     * @param Node[] $nodes
+     */
+    public function __construct(array $nodes = [])
+    {
+        $this->nodes = $nodes;
+    }
+
+    /**
+     * appends a node.
+     *
+     * @param Node|string $node
+     *
+     * @return void
+     */
+    public function append($node)
+    {
+        if (\is_string($node)) {
+            $this->nodes[] = new TextNode($node);
+            return;
+        }
+
+        \assert($node instanceof Node);
+        $this->nodes[] = $node;
+    }
+
+    /**
+     * returns the node-list as array.
+     * this method is mainly used for testing.
+     *
+     * @return array
+     */
+    public function toArray(): array
+    {
+        return $this->nodes;
+    }
+
+    /**
+     * returns the node-list as iterable list.
+     *
+     * (phan has issues with returning an array as iterable)
+     *
+     * @phan-suppress PhanParamSignatureMismatchInternal
+     *
+     * @return Node[]
+     */
+    public function getIterator(): iterable
+    {
+        return $this->nodes;
+    }
```

Comment:
Are you sure phan complains without any good reason here?
I'm getting the following error:
```php
Fatal error: Uncaught Exception: Objects returned by DevCommunityDE\CodeFormatter\Parser\NodeList::getIterator() must be traversable or implement interface Iterator in /srv/app/app/CodeFormatterApp.php:76
```
I'm not sure, but could it be possible that we have to construct and return a specific iterator like [`ArrayIterator`](https://www.php.net/manual/de/arrayiterator.construct.php)?

```suggestion
    /**
     * returns the node-list as iterable list.
     *
     * @return Node[]
     */
    public function getIterator(): iterable
    {
        return new ArrayIterator($this->nodes);
    }
```